crates/next-api/src/app.rs

```diff
@@ -1245,22 +1252,52 @@ impl AppEndpoint {
 
                 let EntryChunkGroupResult {
                     asset: rsc_chunk, ..
-                } = *{
-                    let _span = tracing::trace_span!("server node entrypoint").entered();
-                    chunking_context.entry_chunk_group(
-                        server_path.join(
-                            format!(
-                                "app{original_name}.js",
-                                original_name = app_entry.original_name
-                            )
-                            .into(),
-                        ),
-                        app_entry.rsc_entry,
-                        Vc::cell(evaluatable_assets),
-                        Value::new(AvailabilityInfo::Root),
-                    )
+                } = *(async {
+                    let mut current_chunks = OutputAssets::empty();
+                    let mut current_availability_info = AvailabilityInfo::Root;
+                    if let Some(client_references) = client_references {
+                        for server_component in
+                            client_references.server_component_entries.iter().copied()
+                        {
+                            let server_path = server_component.server_path();
+                            let is_layout =
+                                server_path.file_stem().await?.as_deref() == Some("layout");
```

Comment:
I think that's not even needed. You could just do it for any layout segment. Could you try that?

Reply:
The server components list might be
```
app/layout.js
app/not-found.js
app/sub/layout.js
app/sub/page.js
```
and we only want to inherit modules via `AvailabilityInfo` from parent layouts, not from `not_found`.